crates/engine/tree/src/tree/mod.rs

```diff
@@ -2693,13 +2729,46 @@ where
         } else {
             // The block attaches to canonical persisted parent.
             debug!(target: "engine::tree", %parent_hash, "Parent found on disk");
-            let revert_state = consistent_view.revert_state(parent_hash)?;
+            let revert_state = self.revert_state(provider, best_block_number, parent_hash)?;
             input.append(revert_state);
         }
 
         Ok(input)
     }
 
+    /// Retrieve revert hashed state down to the given block hash.
+    fn revert_state(
+        &self,
+        provider: P::Provider,
+        best_block_number: BlockNumber,
```

Comment:
can we not just use the provider method for best block num here like we did in the prev consistent view method, and remove the argument? given the provider_ro should be greated already

Reply:
we'd query the best block number again, why not pass it as an argument if we already have it?